### PR TITLE
Tweaks to RuntimeVersionCache

### PR DESCRIPTION
**Pull request by dvdplm**
Did some benchmarking and general tinkering and tweaked a few minor things while at it.

Builds on #160 so best look at that first.

### CODE REVIEW

**Review by insipx — APPROVED** [review]
cool, LGTM!

File: substrate-archive-backend/src/runtime_version_cache.rs
Comment (emostov) [inline substrate-archive-backend/src/runtime_version_cache.rs:123]: dq: are these exactly equivalent?

**Review by emostov — APPROVED** [review]
lgtm modulo a nit I am not sure about concerning `RangeBounds` vs using boolean logic (see comment)

File: substrate-archive-backend/src/runtime_version_cache.rs
Comment (emostov) [inline substrate-archive-backend/src/runtime_version_cache.rs:181]: Does using the range have space or time complexity overhead relative to boolean logic?

What not use `b >= self.start && b <= self.end`?

**Review by emostov — COMMENTED** [review]
lgtm modulo a nit I am not sure about concerning `RangeBounds` vs using boolean logic (see comment)

File: substrate-archive-backend/src/runtime_version_cache.rs
Comment (dvdplm) [reply substrate-archive-backend/src/runtime_version_cache.rs:181]: Under the hood they should be exactly the same. 
The idea here is if someone someday figures out a way to optimize `b >= self.start && b <= self.end` we'll get that optimization for free as soon as it lands in the stdlib/compiler. Likely moot in this case, but you never know!

Comment (dvdplm) [reply substrate-archive-backend/src/runtime_version_cache.rs:181]: (It also makes clippy STFU)

File: substrate-archive-backend/src/runtime_version_cache.rs
Comment (dvdplm) [reply substrate-archive-backend/src/runtime_version_cache.rs:123]: They are, yes. It's because of how `AsRef` is defined for `Vec<T>` and `[T]` (I think; others can correct me).

**Pull request merged**
